### ISSUE TITLE
add `weechat.look.prefix_same_nick_wrap`

### ISSUE THREAD

**Issue by Al-Caveman** [weechat/weechat#1563]
## Feature description
add `weechat.look.prefix_same_nick_wrap`, such that:

```
/set weechat.look.prefix_same_nick "└─"
/set weechat.look.prefix_same_nick_middle "├─"
/set weechat.look.prefix_same_nick_wrap "│ "
```

would give a neat look where the green lines are filled properly as follows:

![moarneat](https://i.imgur.com/0UO5FKR.png)

i know we cannot add those green lines now, because we lack `weechat.look.prefix_same_nick_wrap`, hence this feature request.

**Comment by sim642** [comment]
I'm not sure if there is any earlier issue about this because this has definitely been discussed and at some point I put thought into it. I thought I did some testing on this but cannot find it right now.
This isn't too simple to do right now I think because it would make the prefix of a line in the chat somehow be multiple lines while also depending on the wrapping of the message itself.

But besides the technical challenge to the current codebase, there's another one about what would be the right solution. Because just like you want the wrapped lines there to have a connected line in the prefix, someone might want the last wrapped line (where you have written your green text) to also not end at the beginning of the last message but at the end of it (its last wrapped line, going down two more lines). This would split the current single option into _at least_ three:
1. The current one, which is for the first line of the wrapping.
2. The suggested one, which is for the middle lines of the wrapping.
3. The new one, which is for the last line of the wrapping.

But that's not all: since we already have two options for the same nick prefixes (middle and last), for full configurability we'd need the same three for both cases, so at least 6. And if we're already going for wrapped lines with the first-middle-last pattern of options, it wouldn't be unreasonable to also want special configuration for the first same nick line, in addition to the 6 for middle and last same nick lines (wrapped or unwrapped).

And if you think that's all, you'd be wrong. How three options for one wrapped line should be used depending on how many lines it wraps? If there's one line, some people might choose theirs such that option 1 is suitable to mark the beginning of the wrapped text, others might choose theirs such that option 3 is suitable to mark the end of the wrapped text, while neither wants special tickmarks at both first and last line. Similar deal with wrapping just two lines.

In conclusion, my point is that if such feature is added, we should already be planning way ahead and account for all desired use cases. This is to avoid chaos and inconsistency introduced by adding such options one by one to somehow hack around the others, which is how the same nick prefix options have currently come about.

**Comment by Al-Caveman** [comment]
how about a multiple line prefix, with a single variable that has conditioned formatting?  e.g.:
```
/set weechat.look.prefix_line "if:${is_first_line}?${time}${fill}${nick}:if:${is_wrapped}&&${is_middle}?... :if:!${is_wrapped} && ${is_middle}?├─:if:${is_last}?└─"
```

of course just an example, i'm not sure what's the best syntax.  but the idea is:  why not make it into a formatting option with conditional variables in it?

**Comment by Al-Caveman** [comment]
alternatively, how about this:

> there is only 1 true prefix of wrapped lines, and it goes as follows:
>
> - prefix gets added only when the wrapped line does not belong to the
> last line.

?

or, 3 is not bad imo.  e.g. to achieve my effect:

```
/set weechat.look.prefix_same_nick_wrap_first "│ "
/set weechat.look.prefix_same_nick_wrap_middle "│ "
/set weechat.look.prefix_same_nick_wrap_last ""
```

which is backwards compatible if:
```
/set weechat.look.prefix_same_nick_wrap_first ""
/set weechat.look.prefix_same_nick_wrap_middle ""
/set weechat.look.prefix_same_nick_wrap_last ""
```
